File: specification/network/resource-manager/Microsoft.Network/stable/2025-05-01/examples-java/ExpressRouteCircuitGet.java

/**
 * Samples for ExpressRouteCircuits GetByResourceGroup.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/network/resource-manager/Microsoft.Network/stable/2025-05-01/examples/ExpressRouteCircuitGet.json
     */
    /**
     * Sample code: Get ExpressRouteCircuit.
     * 
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void getExpressRouteCircuit(com.azure.resourcemanager.AzureResourceManager azure) {
        azure.networks().manager().serviceClient().getExpressRouteCircuits().getByResourceGroupWithResponse("rg1",
            "circuitName", com.azure.core.util.Context.NONE);
    }
}

File: specification/network/resource-manager/Microsoft.Network/stable/2024-07-01/examples-js/ExpressRouteCircuitGet.js
const { NetworkManagementClient } = require("@azure/arm-network");
const { DefaultAzureCredential } = require("@azure/identity");
require("dotenv/config");

/**
 * This sample demonstrates how to Gets information about the specified express route circuit.
 *
 * @summary Gets information about the specified express route circuit.
 * x-ms-original-file: specification/network/resource-manager/Microsoft.Network/stable/2024-07-01/examples/ExpressRouteCircuitGet.json
 */
async function getExpressRouteCircuit() {
  const subscriptionId = process.env["NETWORK_SUBSCRIPTION_ID"] || "subid";
  const resourceGroupName = process.env["NETWORK_RESOURCE_GROUP"] || "rg1";
  const circuitName = "circuitName";
  const credential = new DefaultAzureCredential();
  const client = new NetworkManagementClient(credential, subscriptionId);
  const result = await client.expressRouteCircuits.get(resourceGroupName, circuitName);
  console.log(result);
}
